src/blant-utils.c

```diff
@@ -123,25 +123,30 @@ unsigned int L_K_Func(Gint_type Gint) {
 unsigned int L_K_Func(Gint_type Gint) {Apology("L_K_Func() not yet implemented"); return -1;}
 #endif
 
+Gordinal_type L_K_Smaller_Canon_Map(Gint_type Gint) {
+    unsigned char perm[MAX_K];
+    Gint_type canon_value = smaller_canon_map(Gint, _k, perm);
+    return canon_to_ordinal(canon_value, _k);
+}
+
 // Assuming the global variable _k is set properly, go read in and/or mmap the big global
 // arrays related to canonical mappings and permutations.
 void SetGlobalCanonMaps(void)
 {
     int i;
     char BUF[BUFSIZ];
-    assert(3 <= _k && _k <= 8);
+    assert(3 <= _k && _k <= 9);
 #if SELF_LOOPS
     _Bk = (1 <<(_k*(_k+1)/2));
 #else
-    _Bk = (1 <<(_k*(_k-1)/2));
+    _Bk = (1 <<(_k_base*(_k_base-1)/2));
 #endif
     _connectedCanonicals = canonListPopulate(BUF, _canonList, _k, _canonNumEdges);
     _numCanon = _connectedCanonicals->maxElem;
     _numConnectedCanon = SetCardinality(_connectedCanonicals);
     _numOrbits = orbitListPopulate(BUF, _orbitList, _orbitCanonMapping, _orbitCanonNodeMapping, _numCanon, _k);
-    _K = (Gordinal_type*) mapCanonMap(BUF, _K, _k);
-
-    sprintf(BUF, "%s/%s/perm_map%d.bin", _BLANT_DIR, CANON_DIR, _k);
+    _K = (Gordinal_type*) mapCanonMap(BUF, _K, _k_base);
```

Comment:
Same here. This needs to work when _k<8.

Reply:
It works